lasy/utils/refractive_index.py

```diff
@@ -0,0 +1,800 @@
+"""
+refractiveindex.info database parser/client.
+
+Inspired somewhat by https://github.com/toftul/refractiveindex/tree/master
+"""
+
+import json
+import os
+import warnings
+from pprint import pprint
+
+import numpy as np
+import scipy.constants as ct
+import yaml
+from scipy.interpolate import CubicSpline
+
+try:
+    import numdifftools as nd
+
+    have_nd = True
+except ImportError:
+    warnings.warn(
+        "numdifftools not available! "
+        "Using fixed numerical expressions for spectral "
+        "phase expansion calculations."
+    )
+    have_nd = False
+
+known_materials_nk = {
+    "fused silica": ("main", "SiO2", "Malitson"),
+    "BK7": ("popular_glass", "BK7", "SCHOTT"),
+    "air": ("other", "air", "Ciddor"),
+    "hydrogen": ("main", "H2", "Peck"),
+    "helium": ("main", "He", "Mansfield"),
+    "argon": ("main", "Ar", "Peck-15C"),
+    "SF11": ("popular_glass", "SF11", "SCHOTT"),
+    "soda lime glass": ("glass", "soda-lime", "Rubin-clear"),
+    "CaF2": ("main", "CaF2", "Malitson"),
+    "MgF2": ("main", "MgF2", "Li-o"),
+    "Sapphire": ("main", "Al2O3", "Malitson-o"),
+}
+
+known_materials_n2 = {
+    "fused silica": ("main", "SiO2"),
+    "BK7": ("glass", "BK7"),
+    "air": ("other", "air"),
+    "helium": ("main", "He"),
+    "argon": ("main", "Ar"),
+    "CaF2": ("main", "CaF2"),
+    "MgF2": ("main", "MgF2"),
+    "Sapphire": ("main", "Al2O3"),
+}
+
+
+class RefractiveIndexDatabase:
+    """
+    Refractive index database for various materials.
+
+    Class that opens and stores the refractiveindex.info
+    YAML database. The entire database will be downloaded
+    on the first time it is run.
+    """
+
+    __database_version = "2025-02-23"
+
+    def __init__(self, database_path=None, auto_download=False):
+        """
+        Initialise the refractive index database.
+
+        Data will be downloaded, if it does not exist
+        and download is requested.
+
+        Parameters
+        ----------
+        database_path : str or None
+            Is None, defaults to user home directory. If passed,
+            should be the directory containing the database
+            structure.
+
+        auto_download : bool, default is True
+            If True, database will be downloaded. If False and
+            no database found, an error will be thrown.
+        """
```

Comment:
Just moved docstring from inside __init__ to class description. We do this generally in LASY, this helps with docs rendering.

```suggestion
    """
    Refractive index database for various materials.

    Class that opens and stores the refractiveindex.info
    YAML database. The entire database will be downloaded if it does not exist and download is requested (typically on the first time it is run).

    Parameters
    ----------
    database_path : str or None
        Is None, defaults to user home directory. If passed,
        should be the directory containing the database
        structure.

    auto_download : bool, default is True
        If True, database will be downloaded. If False and
        no database found, an error will be thrown.
    """

    __database_version = "2025-02-23"

    def __init__(self, database_path=None, auto_download=False):
```